addons/product/migrations/13.0.1.2/pre-migration.py

```diff
@@ -88,6 +88,39 @@ def insert_missing_product_template_attribute_line(env):
         AND insert_ptal.attribute_id = pav.attribute_id)
     GROUP BY pav.id, insert_ptal.id""",
     )
+    if openupgrade.table_exists(env.cr, "mrp_bom_line"):
```

Comment:
I don't get the relation between product_template_attribute_line and mrp boms, neither the code has any reference to this.

Reply:
It's explained in that commit message. In mrp module, the bom lines also have attributes, and some are missing in a similar way the missing attributes on products case you detected. In order to avoid these missing attributes being missing, we need to make use of mrp_bom_line_product_attribute_value_rel, which plays a very similar role that product_attribute_value_product_product_rel has products. We have tested it and works excellent.